libraries/AP_Filesystem/examples/File_IO/File_IO.cpp

```diff
@@ -0,0 +1,71 @@
+//Simple File I/O example
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Filesystem/AP_Filesystem.h>
+
+//set according to the data size in file
+#define BUFFER_SIZE 50
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
```

Comment:
Usually this is just extern hal

Reply:
just extern hal throwing error when building

Reply:
that's because examples don't have a <vehicle>.cpp where this is usually declared